### PR TITLE
verified that simul coloring works with ScipyOptimizeDriver

### PR DESCRIPTION
**Pull request by naylor-b**
- added a test to check calls to solve_linear in the colored and non-colored cases for scipy SLSQP
- turned embedded coloring example back into an active embedded test and changed it to use ScipyOptimizeDriver SLSQP instead of pyoptsparse with SNOPT.
- updated the 'supports' metadata for ScipyOptimizeDriver to say it support simultaneous derivs.
- added a missing _get_name method to ScipyOptimizeDriver.

### CODE REVIEW

File: openmdao/drivers/scipy_optimizer.py [hunk 1]
```diff
@@ -141,6 +142,17 @@ def __init__(self):
 
         self.cite = CITATIONS
 
+    def _get_name(self):
+        """
+        Get name of current driver.
```
Comment (swryan) [inline openmdao/drivers/scipy_optimizer.py]: I see this is the same docstring as in pyOptSparseDriver, but it seems incorrect... it returns the name of the cuurent optimizer, not the name of the driver (which is `ScipyOptimizeDriver`)

File: openmdao/core/tests/test_coloring.py [hunk 2]
```diff
@@ -71,23 +67,25 @@ def run_opt(color_info=None):
...
+    # nonlinear constraints
     p.model.add_constraint('r_con.g', equals=0)
 
     IND = np.arange(SIZE, dtype=int)
-    ODD_IND = IND[0::2]
-    p.model.add_constraint('theta_con.g', equals=0, indices=ODD_IND)
-    p.model.add_constraint('delta_theta_con.g', equals=0)
+    ODD_IND = IND[0::2]  # all odd indices
+    p.model.add_constraint('theta_con.g', lower=-1e-5, upper=1e-5, indices=ODD_IND)#  equals=0,)
+    p.model.add_constraint('delta_theta_con.g', lower=-1e-5, upper=1e-5) # equals=0)
```
Comment (swryan) [inline openmdao/core/tests/test_coloring.py]: clean up these commented bits? (`# equals=0`)

File: openmdao/core/tests/test_coloring.py [hunk 3]
```diff
@@ -157,7 +270,7 @@ def test_simul_coloring(self):
 
     def test_simul_coloring_example(self):
 
-        from openmdao.api import Problem, IndepVarComp, ExecComp, pyOptSparseDriver
+        from openmdao.api import Problem, IndepVarComp, ExecComp, ScipyOptimizeDriver
         import numpy as np
 
         SIZE = 10
```
Comment (swryan) [inline openmdao/core/tests/test_coloring.py]: should the "must be an even number" comment appear here, in the user documentation version?

File: openmdao/core/tests/test_coloring.py (hunk 2 above)
Comment (naylor-b) [reply openmdao/core/tests/test_coloring.py]: fixed

File: openmdao/core/tests/test_coloring.py [hunk 4]
```diff
@@ -102,15 +100,129 @@ def run_opt(color_info=None):
     return p
 
 
-@unittest.skipUnless(OPTIMIZER == 'SNOPT', "This test requires SNOPT.")
 class SimulColoringTestCase(unittest.TestCase):
 
+    @unittest.skipUnless(OPTIMIZER == 'SNOPT', "This test requires SNOPT.")
+    def test_simul_coloring_snopt(self):
+
+        #note: size must always be an even number!!
```
[the remark this answers is not in the document]
Comment (naylor-b) [reply openmdao/core/tests/test_coloring.py]: fixed

File: openmdao/core/tests/test_coloring.py (hunk 3 above)
Comment (naylor-b) [reply openmdao/core/tests/test_coloring.py]: fixed

File: openmdao/drivers/scipy_optimizer.py (hunk 1 above)
Comment (naylor-b) [reply openmdao/drivers/scipy_optimizer.py]: fixed

**Pull request merged**